src/Tests/Microsoft.NET.Sdk.WorkloadManifestReader.Tests/SdkDirectoryWorkloadManifestProviderTests.cs

```diff
@@ -167,6 +167,241 @@ var sdkDirectoryWorkloadManifestProvider
                                 "ios: 18.0.1/7.0.400");
         }
 
+        [Fact]
+        public void ItUsesManifestsFromWorkloadSet()
+        {
+            Initialize("8.0.200");
+
+            CreateMockManifest(_manifestRoot, "8.0.100", "ios", "11.0.1", true);
+            CreateMockManifest(_manifestRoot, "8.0.100", "ios", "11.0.2", true);
+            CreateMockManifest(_manifestRoot, "8.0.200", "ios", "12.0.1", true);
+
+            CreateMockManifest(_manifestRoot, "8.0.200", "android", "33.0.1", true);
+            CreateMockManifest(_manifestRoot, "8.0.200", "android", "33.0.2-rc.1", true);
+            CreateMockManifest(_manifestRoot, "8.0.200", "android", "33.0.2", true);
+
+            CreateMockManifest(_manifestRoot, "8.0.200-rc.1", "maui", "15.0.1-preview.123", true);
+            CreateMockManifest(_manifestRoot, "8.0.200-rc.2", "maui", "15.0.1-rc.456", true);
+            CreateMockManifest(_manifestRoot, "8.0.200", "maui", "15.0.1", true);
+
+
+            CreateMockWorkloadSet(_manifestRoot, "8.0.200", "8.0.200", """
+                {
+                  "ios": "11.0.2/8.0.100",
+                  "android": "33.0.2-rc.1/8.0.200",
+                  "maui": "15.0.1-rc.456/8.0.200-rc.2"
+                }
+                """);
+
+            var sdkDirectoryWorkloadManifestProvider
+                = new SdkDirectoryWorkloadManifestProvider(sdkRootPath: _fakeDotnetRootDirectory, sdkVersion: "8.0.200", userProfileDir: null);
+
+            GetManifestContents(sdkDirectoryWorkloadManifestProvider)
+                .Should()
+                .BeEquivalentTo("ios: 11.0.2/8.0.100", "android: 33.0.2-rc.1/8.0.200", "maui: 15.0.1-rc.456/8.0.200-rc.2");
+        }
+
+        [Fact]
+        public void ItUsesLatestWorkloadSet()
+        {
+            Initialize("8.0.200");
+
+            CreateMockManifest(_manifestRoot, "8.0.100", "ios", "11.0.1", true);
+            CreateMockManifest(_manifestRoot, "8.0.100", "ios", "11.0.2", true);
+            CreateMockManifest(_manifestRoot, "8.0.200", "ios", "12.0.1", true);
+
+            CreateMockWorkloadSet(_manifestRoot, "8.0.200", "8.0.201", """
+    {
+      "ios": "11.0.2/8.0.100"
+    }
+    """);
+
+            CreateMockWorkloadSet(_manifestRoot, "8.0.200", "8.0.202", """
+    {
+      "ios": "12.0.1/8.0.200"
+    }
+    """);
+
+            var sdkDirectoryWorkloadManifestProvider
+                = new SdkDirectoryWorkloadManifestProvider(sdkRootPath: _fakeDotnetRootDirectory, sdkVersion: "8.0.200", userProfileDir: null);
+
+            GetManifestContents(sdkDirectoryWorkloadManifestProvider)
+                .Should()
+                .BeEquivalentTo("ios: 12.0.1/8.0.200");
+        }
+
+        [Fact]
+        public void ItUsesLatestManifestThatIsNotInWorkloadSet()
```

Comment:
```suggestion
        public void ItUsesLatestManifestForWorkloadsNotInWorkloadSet()
```

The old name is a bit ambiguous --> e.g. it can imply that a newer manifest should be used for a workload in the set even if that manifest is not in the set, which is not true.